connectivity/FEATURE_BLE/include/ble/internal/BLEInstanceBase.h

```diff
@@ -22,12 +24,13 @@
 #define MBED_BLE_DEVICE_INSTANCE_BASE__
 
 #include "ble/BLE.h"
-#include "ble/Gap.h"
-#include "ble/SecurityManager.h"
-#include "ble/GattServer.h"
-#include "ble/GattClient.h"
 
+namespace ble {
 
+class SecurityManager;
+class Gap;
+class GattClient;
+class GattServer;
```

Comment:
```suggestion
```
Presumably not needed as `ble/BLE.h` is included where you declared all the classes in one go.

Reply:
revmoed